File: app/src/main/java/com/redinput/notifblock/AppsAdapter.java
package com.redinput.notifblock;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;
import android.preference.PreferenceManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.Arrays;
import java.util.HashSet;
import java.util.List;

public class AppsAdapter extends RecyclerView.Adapter<AppsAdapter.AppsViewHolder> {

    public static class AppsViewHolder extends RecyclerView.ViewHolder {
        ImageView icon;
        TextView name;
        CheckBox hide;
        OnCheckboxAppChecked listener;

        public AppsViewHolder(View itemView, final OnCheckboxAppChecked listener) {
            super(itemView);

            this.listener = listener;
            icon = (ImageView) itemView.findViewById(R.id.app_icon);
            name = (TextView) itemView.findViewById(R.id.app_name);
            hide = (CheckBox) itemView.findViewById(R.id.app_hide);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    hide.performClick();
                }
            });

            hide.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                @Override
                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                    listener.onCheckboxAppChecked(getLayoutPosition(), isChecked);
                }
            });
        }
    }

    private PackageManager pm;
    private LayoutInflater mLayoutInflater;

    private List<ApplicationInfo> apps;
    private OnCheckboxAppChecked checkedListener;
    private SharedPreferences preferences;

    public AppsAdapter(Context context, List<ApplicationInfo> apps, OnCheckboxAppChecked checkedListener) {
        this.apps = apps;
        pm = context.getPackageManager();
        mLayoutInflater = LayoutInflater.from(context);
        this.checkedListener = checkedListener;
        preferences = PreferenceManager.getDefaultSharedPreferences(context);

        setHasStableIds(true);
    }

    @Override
    public AppsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = mLayoutInflater.inflate(R.layout.app_recycler_row, parent, false);
        return new AppsViewHolder(v, checkedListener);
    }

    @Override
    public void onBindViewHolder(AppsViewHolder holder, int position) {
        ApplicationInfo app = apps.get(position);
        HashSet<String> blocked = new HashSet<>(Arrays.asList(preferences.getString(Utils.PREF_PACKAGES_BLOCKED, "").split(";")));

        holder.icon.setImageDrawable(app.loadIcon(pm));
        holder.name.setText(app.loadLabel(pm));
        holder.hide.setChecked(blocked.contains(app.packageName));
    }

    @Override
    public long getItemId(int position) {
        return (long) apps.get(position).packageName.hashCode();
    }

    @Override
    public int getItemCount() {
        return apps.size();
    }

    public ApplicationInfo getIem(int position) {
        return apps.get(position);
    }

    public void setApps(List<ApplicationInfo> apps) {
        this.apps = apps;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/redinput/notifblock/AppsAdapter.kt
package com.redinput.notifblock

import android.content.Context
import android.content.SharedPreferences
import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager
import android.preference.PreferenceManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.redinput.notifblock.AppsAdapter.AppsViewHolder
import java.util.*

class AppsAdapter(context: Context, private var apps: List<ApplicationInfo>, checkedListener: OnCheckboxAppChecked) : RecyclerView.Adapter<AppsViewHolder>() {

    class AppsViewHolder(itemView: View, var listener: OnCheckboxAppChecked) : RecyclerView.ViewHolder(itemView) {
        var icon: ImageView
        var name: TextView
        var hide: CheckBox

        init {
            icon = itemView.findViewById(R.id.app_icon)
            name = itemView.findViewById(R.id.app_name)
            hide = itemView.findViewById(R.id.app_hide)
            itemView.setOnClickListener { hide.performClick() }
            hide.setOnCheckedChangeListener { buttonView, isChecked -> listener.onCheckboxAppChecked(layoutPosition, isChecked) }
        }
    }

    private val pm: PackageManager
    private val mLayoutInflater: LayoutInflater
    private val checkedListener: OnCheckboxAppChecked
    private val preferences: SharedPreferences
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AppsViewHolder {
        val v = mLayoutInflater.inflate(R.layout.app_recycler_row, parent, false)
        return AppsViewHolder(v, checkedListener)
    }

    override fun onBindViewHolder(holder: AppsViewHolder, position: Int) {
        val app = apps[position]
        val blocked = HashSet(Arrays.asList(*preferences.getString(Utils.PREF_PACKAGES_BLOCKED, "")!!.split(";").toTypedArray()))
        holder.icon.setImageDrawable(app.loadIcon(pm))
        holder.name.text = app.loadLabel(pm)
        holder.hide.isChecked = blocked.contains(app.packageName)
    }

    override fun getItemId(position: Int): Long {
        return apps[position].packageName.hashCode().toLong()
    }

    override fun getItemCount(): Int {
        return apps.size
    }

    fun getIem(position: Int): ApplicationInfo {
        return apps[position]
    }

    fun setApps(apps: List<ApplicationInfo>) {
        this.apps = apps
        notifyDataSetChanged()
    }

    init {
        pm = context.packageManager
        mLayoutInflater = LayoutInflater.from(context)
        this.checkedListener = checkedListener
        preferences = PreferenceManager.getDefaultSharedPreferences(context)
        setHasStableIds(true)
    }
}